### ISSUE TITLE
Kitty stops functioning after reconnect if an app which has extended keyboard handling is run

### ISSUE THREAD

**Issue by dredkin** [cyd01/KiTTY#421]
1. Connect to a host
2. Run a program which uses extended keyboard handling (e.g. far2l)
3. Wait for a disconnect (e.g. put the client comp to sleep)
4. wake up from the sleep.
5. Kitty reconnects successfully, but stops working

The command prompt is shown ok, but any input produces garbage, like below:
```
dredkin@redkin-server:~$ f2lAQBYAC0AIAAAAEcEAABLf2lAQBYAC0AIAAAAEcEAABrf2lAQAIAAAAIAAAAAgAAABLf2lAQAIAAAAIAAAAAgAAABLf2lAQAIAAAAIAAAAAgAAABLf2lAQAIAAAAIAAAAAgAAABrf2lAQARAB0AKAAAAAAAAABLf2lAQBDAC4AKAAAAAMAAABLf2lAQBDAC4AKAAAAAMAAABrf2lAQARAB0AIAAAAAAAAABr
```

NB: if you exit the app before disconnect, KiTTy reconnects ok, So this should be some actions which are done when app is closed, but not done before the reconnect.

**Comment by dredkin** [comment]
@mihmig  looks like this is for you.

**Comment by dredkin** [comment]
@cyd01 Could you please assist me? I seem to know how to fix this bug, but I just can't compile Kitty to check it works (the instruction in readme is outdated).

**Comment by unxed** [comment]
It looks as if the term->far2l_ext flag is not reset when the connection is re-established, and after reconnecting kitty continues to send keyboard events in the far2l escape sequence format, which normal bash, of course, cannot recognize

**Comment by unxed** [comment]
Should help

```
diff --git a/0.76b_My_PuTTY/windows/window.c b/0.76b_My_PuTTY/windows/window.c
index 530e2ff..98b9c4a 100644
--- a/0.76b_My_PuTTY/windows/window.c
+++ b/0.76b_My_PuTTY/windows/window.c
@@ -787,9 +787,14 @@ void RestartSession( void ) {
 		win_seat_connection_fatal(  &wgs.seat, "User request session restart..." ) ;
 		SetTimer(wgs.term_hwnd, TIMER_RECONNECT, 10, NULL) ;
 	}
+
+#ifdef MOD_FAR2L
+    term->far2l_ext = 0;
+#endif
 	
 	PostMessage(wgs.term_hwnd,WM_KEYDOWN,VK_RETURN ,0) ;
 	PostMessage(wgs.term_hwnd,WM_KEYUP,VK_RETURN ,1) ;
 }
 #endif
 
```

**Comment by cyd01** [comment]
> @cyd01 Could you please assist me? I seem to know how to fix this bug, but I just can't compile Kitty to check it works (the instruction in readme is outdated).

Here are the compilation instructions: http://www.9bis.net/kitty/#!pages/HowToCompile.md

**Comment by unxed** [comment]
> This should help

The above fix does actually help

**Comment by dredkin** [comment]
#432

**Comment by unxed** [comment]
> #432

Thanks!

**Comment by cyd01** [comment]
I've just released 0.76.0.12 version with your patch.

**Comment by unxed** [comment]
Great, thanks!
